packages/devtools_app/lib/src/screens/network/network_screen.dart

```diff
@@ -207,55 +209,48 @@ class _NetworkProfilerControlsState extends State<_NetworkProfilerControls>
-            gaScreen: gac.network,
-            gaSelection: gac.clear,
-            onPressed: widget.controller.clear,
-          ),
-          const SizedBox(width: defaultSpacing),
-          DownloadButton(
-            minScreenWidthForTextBeforeScaling:
-                _NetworkProfilerControls._includeTextWidth,
-            onPressed: widget.controller.exportAsHarFile,
-            gaScreen: gac.network,
-            gaSelection: gac.NetworkEvent.downloadAsHar.name,
-          ),
-          const SizedBox(width: defaultSpacing),
-          const Expanded(child: SizedBox()),
-          // TODO(kenz): fix focus issue when state is refreshed
-          SearchField<NetworkController>(
-            searchController: widget.controller,
-            searchFieldEnabled: hasRequests,
-            searchFieldWidth: screenWidth <= MediaSize.xs
-                ? defaultSearchFieldWidth
-                : wideSearchFieldWidth,
-          ),
-          const SizedBox(width: denseSpacing),
-          DevToolsFilterButton(
-            onPressed: _showFilterDialog,
-            isFilterActive: _filteredRequests.length != _requests.length,
-          ),
-        ],
+        StartStopRecordingButton(
+          recording: _recording,
+          onPressed: () async =>
+              await widget.controller.togglePolling(!_recording),
+          tooltipOverride: _recording
+              ? 'Stop recording network traffic'
+              : 'Resume recording network traffic',
+          minScreenWidthForTextBeforeScaling: double.infinity,
+          gaScreen: gac.network,
+          gaSelection: _recording ? gac.pause : gac.resume,
+        ),
+        const SizedBox(width: denseSpacing),
+        ClearButton(
+          minScreenWidthForTextBeforeScaling:
+              _NetworkProfilerControls._includeTextWidth,
+          gaScreen: gac.network,
+          gaSelection: gac.clear,
+          onPressed: widget.controller.clear,
+        ),
+        const SizedBox(width: defaultSpacing),
+        DownloadButton(
+          minScreenWidthForTextBeforeScaling:
+              _NetworkProfilerControls._includeTextWidth,
+          onPressed: widget.controller.exportAsHarFile,
+          gaScreen: gac.network,
+          gaSelection: gac.NetworkEvent.downloadAsHar.name,
+        ),
+        const SizedBox(width: defaultSpacing),
+        const Expanded(child: SizedBox()),
+        // TODO(kenz): fix focus issue when state is refreshed
+        SearchField<NetworkController>(
+          searchController: widget.controller,
+          searchFieldEnabled: hasRequests,
+          searchFieldWidth: screenWidth <= MediaSize.xs
+              ? defaultSearchFieldWidth
+              : wideSearchFieldWidth,
+        ),
+        const SizedBox(width: denseSpacing),
+        DevToolsFilterButton(
+          onPressed: _showFilterDialog,
+          isFilterActive: _filteredRequests.length != _requests.length,
+        ),
```

Comment:
only include these two controls in offline mode.

```
children: [
  if(!widget.offline) ...[
    StartStopRecordingButton(...)
    const SizedBox(width: denseSpacing),
    ClearButton(...)
    const SizedBox(width: denseSpacing),
    DownloadButton(...)
    const SizedBox(width: denseSpacing),
  ]
  Spacer(), // this is a helper widget that replaces Expanded(child: SizedBox()),
  const SizedBox(width: denseSpacing),
  SearchField<NetworkController>(...)
  const SizedBox(width: denseSpacing),
  DevToolsFilterButton(...),
],
```

Reply:
done